### PR TITLE
Remove itwin-client dependency and use axios

### PR DESCRIPTION
**Pull request by calebmshafer**
Update to the latest RC version of the packages and drop the use of itwin-client.

Plan to release this as 0.4.0 to push out the axios update before marking it as 1.0.0.

### CODE REVIEW

File: src/ProjectsClient.ts [hunk 1]
```diff
@@ -39,21 +39,21 @@ export class ProjectsAccessClient implements ProjectsAccess {
...
+    const requestOptions = this.getRequestOptions(accessToken);
     let url = this._baseUrl;
     if (queryArg)
       url = url + this.getQueryString(queryArg);
 
     const projects: Project[] = [];
 
     try {
-      const response = await request(url, requestOptions);
+      const response = await axios.get(url, requestOptions);
```
Comment (aruniverse) [inline src/ProjectsClient.ts:50]: couldn't we just use fetch?

Comment (calebmshafer) [reply src/ProjectsClient.ts:50]: wanted to make sure it would work in node/backend code as well.

Comment (aruniverse) [reply src/ProjectsClient.ts:50]: ah gotcha

**Review by aruniverse — APPROVED** [review]

File: src/ProjectsClient.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@
  * @module ProjectsClient
  */
 import type { AccessToken } from "@itwin/core-bentley";
-import { request, RequestOptions } from "@bentley/itwin-client";
+import axios, { AxiosRequestConfig } from "axios";
```
Comment (SpencerWBarnes) [inline src/ProjectsClient.ts:9]: Why is axios outside the import braces? I'm not familiar with this format

File: package.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
     "url": "https://github.com/iTwin/projects-client"
   },
   "dependencies": {
-    "@bentley/itwin-client": "3.0.0-dev.138"
+    "axios": "^0.25.0"
```
Comment (SpencerWBarnes) [inline package.json:36]: Is there any risk in relying on a library in version 0? And yes I see the irony that this package is version 0 as well.

File: src/ProjectsClient.ts (hunk 2 above)
Comment (aruniverse) [reply src/ProjectsClient.ts:9]: axios is the default import, named imports are inside the braces

File: package.json (hunk 3 above)
Comment (aruniverse) [reply package.json:36]: yea is def a risk, but axios is a very popular lib and should be ok for this

File: src/ProjectsClient.ts (hunk 2 above)
Comment (SpencerWBarnes) [reply src/ProjectsClient.ts:9]: Oh okay, thanks

File: package.json (hunk 3 above)
Comment (SpencerWBarnes) [reply package.json:36]: I hadn't heard of it before, but maybe I should look into it because it seems very useful

**Review by SpencerWBarnes — APPROVED** [review]

**Pull request merged**